### PR TITLE
CSI-2778 reducing controller server length

### PR DESCRIPTION
**Pull request by zingero**

### CODE REVIEW

File: controller/controller_server/csi_controller_server.py [hunk 1]
```diff
@@ -283,54 +260,24 @@ def ControllerPublishVolume(self, request, context):
...
+        utils.validate_unpublish_volume_request(request)
+
         try:
-            try:
-                utils.validate_unpublish_volume_request(request)
-            except ValidationException as ex:
-                logger.exception(ex)
-                context.set_details(ex.message)
-                context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
-                return csi_pb2.ControllerUnpublishVolumeResponse()
```
Comment (oriyarde) [inline controller/controller_server/csi_controller_server.py:281]: I think this empty line should be removed as well

File: controller/controller_server/csi_controller_server.py [hunk 2]
```diff
@@ -485,46 +394,25 @@ def DeleteSnapshot(self, request, context):
...
 
+    @handle_common_exceptions(csi_pb2.GetCapacityResponse)
     def GetCapacity(self, request, context):
         logger.info("GetCapacity")
-        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
-        logger.info("finished GetCapacity")
-        return csi_pb2.GetCapacityResponse()
+        raise NotImplementedError()
 
+    @handle_common_exceptions(csi_pb2.ControllerExpandVolumeResponse)
```
Comment (oriyarde) [inline controller/controller_server/csi_controller_server.py:415]: so could we move the `validate_expand_volume_request` call to be outside the `try`?

**Review by oriyarde — DISMISSED** [review]
LGTM (not for this version though)

File: controller/controller_server/csi_controller_server.py (hunk 1 above)
Comment (zingero) [reply controller/controller_server/csi_controller_server.py]: Done

File: controller/controller_server/csi_controller_server.py (hunk 2 above)
Comment (zingero) [reply controller/controller_server/csi_controller_server.py:407]: Done

**Review by oriyarde — APPROVED** [review]
LGTM

**Review by ArbelNathan — APPROVED** [review]
great job!

File: controller/controller_server/csi_controller_server.py [hunk 3]
```diff
@@ -350,57 +291,31 @@ def ControllerUnpublishVolume(self, request, context):
 
             logger.info("finished ControllerUnpublishVolume")
             return csi_pb2.ControllerUnpublishVolumeResponse()
-
+        except ObjectIdError as ex:
+            return handle_exception(ex, context, grpc.StatusCode.INVALID_ARGUMENT,
+                                    array_errors.VolumeAlreadyUnmappedError)
```
Comment (oriyarde) [inline controller/controller_server/csi_controller_server.py:296]: an array error is not a response type

Comment (zingero) [reply controller/controller_server/csi_controller_server.py:296]: @oriyarde Do you want me to change it?

Comment (oriyarde) [reply controller/controller_server/csi_controller_server.py:296]: handled in https://github.com/IBM/ibm-block-csi-driver/pull/496

**Pull request merged**